homeassistant/components/calendar/__init__.py

```diff
@@ -7,179 +7,200 @@
 import asyncio
 import logging
 from datetime import timedelta
-import re
 
-from homeassistant.components.google import (
-    CONF_OFFSET, CONF_DEVICE_ID, CONF_NAME)
-from homeassistant.const import STATE_OFF, STATE_ON
-from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
-from homeassistant.helpers.config_validation import time_period_str
-from homeassistant.helpers.entity import Entity, generate_entity_id
-from homeassistant.helpers.entity_component import EntityComponent
-from homeassistant.helpers.template import DATE_STR_FORMAT
-from homeassistant.util import dt
+from aiohttp.web_exceptions import HTTPNotFound
 
-_LOGGER = logging.getLogger(__name__)
 
-DOMAIN = 'calendar'
+import homeassistant.util.dt as dt
+
+from homeassistant.helpers import config_per_platform, discovery
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.setup import async_prepare_setup_platform
 
-ENTITY_ID_FORMAT = DOMAIN + '.{}'
+from homeassistant.const import STATE_OFF, STATE_ON
 
-SCAN_INTERVAL = timedelta(seconds=60)
+DEPENDENCIES = ['http']
+DOMAIN = 'calendar'
+SCAN_INTERVAL = timedelta(seconds=10)
+_LOGGER = logging.getLogger(__name__)
 
 
 @asyncio.coroutine
 def async_setup(hass, config):
     """Track states and offer events for calendars."""
-    component = EntityComponent(
-        _LOGGER, DOMAIN, hass, SCAN_INTERVAL, DOMAIN)
+    calendars = []
 
-    yield from component.async_setup(config)
-    return True
+    component = EntityComponent(_LOGGER, DOMAIN, hass, SCAN_INTERVAL)
+
+    hass.components.frontend.register_built_in_panel(
+        'calendar', 'Calendar', 'mdi:calendar')
+    hass.http.register_view(CalendarPlatformsView(calendars))
+    hass.http.register_view(CalendarEventView(calendars))
+
+    @asyncio.coroutine
+    def async_setup_platform(p_type, p_config=None, discovery_info=None):
```

Comment:
Why do we need this here? Let platforms implement it. That's what `EntityComponent` expects.